### PR TITLE
A bunch of re-organization followed by adding a bunch of code

### PR DESCRIPTION
**Pull request by tfenne**
The commits are stepwise, and it's probably easiest to review commit by commit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,12 +53,6 @@ conda activate fgpyo
 poetry install
 ```
 
-If, during `poetry install` on Mac OS X errors are encountered running gcc/clang to build `pybedtools` or other packages with native code, try setting the following and re-running `poetry install`:
-```bash
-export CFLAGS="-stdlib=libc++"
-``` 
-
-
```
Comment (nh13) [inline README.md:61]: why is this hint no longer needed?

File: fgpyo/collections/__init__.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+from typing import Any
+from typing import Callable
+from typing import Generic
+from typing import Iterable
+from typing import Iterator
+from typing import List
+from typing import TypeVar
+from typing import Union
+
+
```
Comment (nh13) [inline fgpyo/collections/__init__.py:10]: 1. I think we should give attribution to where this came from
2. I think the long docs with examples are super useful to new folks on our team to show how to use this like you did for the other modules.  Can you copy those over too?

File: fgpyo/collections/tests/test_peekable_iterator.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+"""Tests for :py:mod:`~samwell.itertools`"""
```
Comment (nh13) [inline fgpyo/collections/tests/test_peekable_iterator.py:1]:
```suggestion
"""Tests for :py:mod:`~fgpyo.collections.PeekableIterator`"""
```

File: fgpyo/sam/__init__.py [hunk 4]
```diff
@@ -0,0 +1,586 @@
...
+    """
+
+    elements: Tuple[CigarElement, ...] = attr.ib(default=())
+
+    @classmethod
+    def from_cigartuples(cls, cigartuples: Optional[List[Tuple[int, int]]]) -> "Cigar":
+        """Returns a Cigar from a list of tuples returned by pysam.
+
+        Each tuple denotes the operation and length.  See
+        :class:`~samwell.sam.CigarOp` for more information on the
```
Comment (nh13) [inline fgpyo/sam/__init__.py:417]:
```suggestion
        :class:`~fgpyo.sam.CigarOp` for more information on the
```

**Review by nh13 — APPROVED** [review]
Minor stuff. I do think we should add attribution to where these came from.  Thank-you for doing this!!!

File: README.md (hunk 1 above)
Comment (tfenne) [reply README.md:61]: Because pybedtools is no longer a [transitive] dependency!

File: fgpyo/collections/__init__.py (hunk 2 above)
Comment (tfenne) [reply fgpyo/collections/__init__.py:71]: I thought I had copied those module docs - have now.
Re: attribution I added a section to the LICENSE doc.  I think that's sufficient to merge but if you feel like we should add more elsewhere let me know.

**Pull request merged**